java/core/src/java/org/apache/orc/impl/InStream.java

```diff
@@ -397,13 +397,14 @@ public String toString() {
     }
   }
 
-  private static class CompressedStream extends InStream {
+  static class CompressedStream extends InStream {
     private final int bufferSize;
     private ByteBuffer uncompressed;
     private final CompressionCodec codec;
     protected ByteBuffer compressed;
     protected DiskRangeList currentRange;
     private boolean isUncompressedOriginal;
+    protected long currentCompressedStart;
```

Comment:
This should be initialized to -1 so that we don't have someone seek to 0 and get incorrect data.

Reply:
Fixed